packages/config/src/index.ts

```diff
@@ -56,14 +57,24 @@ export async function loadConfig<U extends UserConfig>(
     defaults: inlineConfig,
   })
 
-  const result = await loader.load()
-  result.config = Object.assign(defaults, result.config || inlineConfig)
-  if (result.config.configDeps) {
-    result.sources = [
-      ...result.sources,
-      ...result.config.configDeps.map(i => resolve(cwd, i)),
-    ]
-  }
+  try {
+    const result = await loader.load()
+    result.config = Object.assign(defaults, result.config || inlineConfig)
+    if (result.config.configDeps) {
+      result.sources = [
+        ...result.sources,
+        ...result.config.configDeps.map(i => resolve(cwd, i)),
+      ]
+    }
+    historyLoadedConfig = result
 
-  return result
+    return result
+  }
+  catch (e) {
+    if (historyLoadedConfig) {
```

Comment:
I don't think this is a robust approach. `loadConfig` should be side-effect free if possible. Also unconditionally cache the previous config would lead to tricky bugs as the arguments could be different, and the cache might be shared across different calls.

I would suggest move this handling to the higher-level abstraction, like `createCachedConfigLoader` where we save the cache inside the instance and the behavior could be more explicit